File: fw-portal-web/src/main/resources/static/resources/js/dto/CargaDto.js
class CargaDto{
    _valor;
    _produtoPredominante;
    _medidas=[];

    constructor(valor,produtoPredominante,medidas){
        this._valor = valor;
        this._produtoPredominante = produtoPredominante;
        this._medidas = medidas.map(function(object){
            return new MedidaDto(object.tipoMedida,object.quantidade);
        });
    }

    getValor(){
        return this._valor;
    }

    getProdutoPredominante(){
        return this._produtoPredominante;
    }
    getMedidas(){
        return this._medidas;
    }

}

File: fw-portal-web/src/main/java/com/faraway/fwportal/dto/CargaDto.java
package com.faraway.fwportal.dto;

import java.math.BigDecimal;
import java.util.HashSet;
import java.util.Set;
import java.util.concurrent.ConcurrentSkipListSet;

import com.faraway.fwportal.model.domain.Carga;

public class CargaDto {

	private BigDecimal valor;

	private String produtoPredominante;

	private Set<MedidaDto> medidas;

	public CargaDto(Carga carga) {
		super();
		this.valor = carga.getValor();
		this.produtoPredominante = carga.getProdutoPredominante();
		this.medidas = carga.getMedidas().parallelStream().map(MedidaDto::new).collect(ConcurrentSkipListSet::new,
				Set::add, Set::addAll);
	}

	public BigDecimal getValor() {
		return valor;
	}

	public String getProdutoPredominante() {
		return produtoPredominante;
	}

	public Set<MedidaDto> getMedidas() {
		return new HashSet<>(medidas);
	}

}
